service/src/main/java/greencity/mapping/CustomHabitMapper.java

```diff
@@ -0,0 +1,19 @@
+package greencity.mapping;
+
+import greencity.dto.habit.AddCustomHabitDtoRequest;
+import greencity.entity.Habit;
+import org.modelmapper.AbstractConverter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CustomHabitMapper extends AbstractConverter<AddCustomHabitDtoRequest, Habit> {
+    @Override
+    public Habit convert(AddCustomHabitDtoRequest addCustomHabitDtoRequest) {
+        return Habit.builder()
+            .image(addCustomHabitDtoRequest.getImage())
+            .complexity(addCustomHabitDtoRequest.getComplexity())
+            .defaultDuration(addCustomHabitDtoRequest.getDefaultDuration())
+            .isCustomHabit(true)
```

Comment:
Should we add `user_id` too for custom habit?

Reply:
Why cant we set `user_id` here?

Reply:
I changed parameters in method in controller. Now we find the user from the principals and in order to set userId we use UserRepo userRepo.findByEmail(userEmail)